core/services/chainlink/config_general.go

```diff
@@ -135,6 +137,8 @@ func (o GeneralConfigOpts) New() (GeneralConfig, error) {
 		return nil, err
 	}
 
+	_, warning := utils.MultiErrorList(o.Config.deprecationWarnings())
```

Comment:
I originally tried to hook in to validation, but that doesn't make sense since the defaults have already been applied at that point. The only time to do this check is right here, after merging all user inputs, and before setting the defaults.